File: solutions/fsharp/space-age/SpaceAge.fs
module SpaceAge

type Planet = | Earth | Mercury | Venus | Mars | Jupiter | Saturn | Uranus | Neptune

let rec age (planet: Planet) (seconds: int64): float =
    let earthAge = float seconds / 31557600.0
    match planet with
    | Earth -> earthAge
    | Mercury -> earthAge / 0.2408467
    | Venus -> earthAge / 0.61519726
    | Mars ->  earthAge / 1.8808158
    | Jupiter ->  earthAge / 11.862615
    | Saturn ->  earthAge / 29.447498
    | Uranus ->  earthAge / 84.016846
    | Neptune ->  earthAge / 164.79132

File: solutions/python/space-age/space_age.py
"""Compute years on various planets."""

EARTH_YEAR_IN_SECONDS = 31_557_600

ORBIT_RATIOS = (
    ('earth', 1.0),
    ('mercury', 0.2408467),
    ('venus', 0.61519726),
    ('mars', 1.8808158),
    ('jupiter', 11.862615),
    ('saturn', 29.447498),
    ('uranus', 84.016846),
    ('neptune', 164.79132)
)


class SpaceAge:
    """Compute years on various planets."""

    def __init__(self, seconds):
        for planet, ratio in ORBIT_RATIOS:
            age_in_years = round(seconds / EARTH_YEAR_IN_SECONDS / ratio, 2)
            setattr(self, 'on_' + planet,lambda x=age_in_years: x)
